src/aks-preview/azext_aks_preview/aks_app_commands/aksappcommands.py

```diff
@@ -0,0 +1,165 @@
+    architecture = platform.machine().lower()
+
+    if architecture == 'x86_64':
+        architecture = 'amd64'
+    if architecture not in ['arm64', 'amd64']:
+        print('Cannot find a suitable download for the current system architecture. Draft only supports AMD64 and ARM64.')
+        return None
+
+    return f'draftv2-{operating_system}-{architecture}'
+
+
+# Returns path to existing draftv2 binary and None otherwise
+def _get_existing_path() -> Optional[str]:
+    print('Checking if DraftV2 binary exists locally...')
+
+    filename = _get_filename()
+    if not filename:
+        return None
+
+    paths = _get_potential_paths()
+    if not paths:
+        print('List of potential DraftV2 paths is empty')
+        return None
+
+    for path in paths:
+        binary_file_path = path + '/' + filename
+        if os.path.exists(binary_file_path):
+            print('Existing binary found at: ' + binary_file_path)
+            return binary_file_path
+    return None
+
+
+# Returns a list of potential draftV2 binary paths
+def _get_potential_paths() -> List[str]:
+    result = [str(Path.home()) + '/' + '.aksapp']
+    paths = os.environ['PATH'].split(':')
+
+    for path in paths:
+        if 'draftv2' in path:
+            result.append(path)
+    return result
+
+
+def _build_arguments(app_name: str,
+                     language: str,
+                     create_config: str,
+                     dockerfile_only: str,
+                     deployment_only: str) -> List[str]:
+    options = {
+        'app-name': app_name,
+        'language': language,
+        'create-config': create_config,
+        'dockerfile-only': dockerfile_only,
+        'deployment-only': deployment_only
+    }
+    args_list = []
+    for arg, val in options.items():
+        if val:
+            args_list.append(f'--{arg}={val}')
+    return args_list
+
+
+def _run(binary_path: str, deployment_path: str, arguments: List[str]) -> bool:
+    if binary_path is None:
+        raise ValueError('The given Binary path was null or empty')
+
+    print('Running DraftV2 Binary ...')
+    cmd = [binary_path, 'create', deployment_path] + arguments
+    process = subprocess.Popen(cmd)
+    exit_code = process.wait()
+    return exit_code == 0
+
+
+def _cmd_finish():
+    # Clean up logic can go here if needed
+    print('Finishing running \'az aks app init\'')
+
+
+def _download_binary() -> Optional[str]:
+    # prompt user to download binary. If users says no, we error out and tell them that this requires the binary
+    msg = 'The required binary was not found. Would you like us to download the required binary for you?'
+
+    if not prompt_y_n(msg, default='n'):
+        raise ValueError('`az aks app` requires the missing dependency')
+
+    print('Attempting to download dependency...')
+
+    filename = _get_filename()
+    if not filename:
+        return None
+
+    url = f'https://github.com/Azure/aks-app/releases/latest/download/{filename}'
```

Comment:
I think we want to download a specific version of aks-app here. That way you can control when customers get updates

Reply:
We had a discussion on this during the first iteration and the solution was to use the latest release for now and potentially change to a specific tag later. more info [here](https://github.com/Azure/azure-cli-extensions/pull/4651#discussion_r846028241)